agent/utils/retry/retry.go

```diff
@@ -15,13 +15,28 @@ package retry
 
 import (
 	"context"
+	"sync"
+	"time"
 
 	apierrors "github.com/aws/amazon-ecs-agent/agent/api/errors"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/logger"
 	"github.com/aws/amazon-ecs-agent/agent/utils/ttime"
 )
 
 var _time ttime.Time = &ttime.DefaultTime{}
 
+type TaskEventsFlowController struct {
+	flowControl      map[string]chan bool
+	eventControlLock sync.RWMutex
```

Comment:
nit: this is a RWMutex but I don't see anywhere we're using read-lock only, so maybe just leave it as sync.Mutex, later code changes could change it to RWMutex if a read-lock is needed or valid somewhere.